cmd/abapEnvironmentCheckoutBranch.go

```diff
@@ -150,3 +146,85 @@ func triggerCheckout(repositoryName string, branchName string, checkoutConnectio
 	uriConnectionDetails.URL = body.Metadata.URI + expandLog
 	return uriConnectionDetails, nil
 }
+
+func checkCheckoutBranchRepositoryConfiguration(options abapEnvironmentCheckoutBranchOptions) error {
+	if options.Repositories == "" && options.RepositoryName == "" && options.BranchName == "" {
+		return fmt.Errorf("Checking configuration failed: %w", errors.New("You have not specified any repository configuration to be pulled into the ABAP Environment System. Please make sure that you specified the repositories with their branches that should be pulled either in a dedicated file or via in-line configuration. For more information please read the User documentation"))
+	}
+	if options.Repositories != "" && options.RepositoryName != "" && options.BranchName != "" {
+		log.Entry().Info("It seems like you have specified both the repositories with their branches to be pulled as an in-line configuration as well as in the dedicated repositories configuration file.")
+		log.Entry().Info("Please note that in this case the dedicated repositories configuration file will be handled with priority.")
+	}
+	if options.Repositories != "" && ((options.RepositoryName == "") != (options.BranchName == "")) {
+		log.Entry().Info("It seems like you have specified a dedicated repository configuration file but also an in-line configuration for the repository or branch to be pulled.")
+		log.Entry().Info("Please note that in this case the dedicated repositories configuration file will be handled only.")
+	}
+	return nil
+}
+
+func checkoutBranchesFromConfigFile(repositoriesFile string, checkoutConnectionDetails abaputils.ConnectionDetailsHTTP, client piperhttp.Sender, pollIntervall time.Duration) (err error) {
+	if fileExists(repositoriesFile) {
```

Comment:
How about reversing the if here and return early with the error from the else path? Then you can keep a consistent happy path indentation.

Reply:
This function has been removed and replaced by `getRepositories` function in `abaputils` pkg. File handling is done there including error handling. This function returns a list of repositories from reading & handling this file and/or step parameters.

Should make things more readable and better maintainable.